sel/test/constant_test.cpp

```diff
@@ -0,0 +1,59 @@
+#include "sel/constant.hpp"
+#include "skytest/skytest.hpp"
+
+#include <compare>
+#include <cstddef>
+#include <format>
+#include <limits>
+
+auto main() -> int
+{
+  using namespace skytest::literals;
+  using namespace skytest;
+
+  "constant value cannot be unordered"_test = [] {
+    static constexpr auto nan = []<class T>(T) {
+      return std::numeric_limits<T>::quiet_NaN();
+    };
+    return expect(
+        aborts([] { sel::constant{nan(float{})}; }) and
+        aborts([] { sel::constant{nan(double{})}; })
```

Comment:
just out of curiosity, is this passing an instance of float/double? couldn't you write nan<float>() or nan<double>() or

Reply:
`nan(float{})` returns a `float` with a NaN-value and `nan(double{})` returns a `double` with a NaN-value.

I guess we could use these: https://en.cppreference.com/w/cpp/numeric/math/nan.html
since this can't be a constexpr test since we are testing `abort`.

Is that what you're suggesting?

Reply:
No I'm just confused by the syntax. I thought `float{}` would create a (default) instance of float (0.0f) and the only reason that's being done is to select the generic type parameter.

Reply:
> I thought float{} would create a (default) instance of float (0.0f) and the only reason that's being done is to select the generic type parameter.

Yeah - can switch to using `std::numeric_limits<T>::quiet_Nan()` as well

Reply:
I would have defined a variable template (e.g. `nan_v<float>`) but you can't define those at function scope.

I'll just use numeric_limits directly.

Reply:
https://github.com/digiboys/sel/compare/527dd5622d4799e454cd32851cb610207eea7d90..d51eed59fb15c6f0903d4b52309edbaf1d5c6ee5